irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -85,7 +85,17 @@ namespace iroha {
           }
           auto chain = blockLoader_->retrieveBlocks(
               shared_model::crypto::PublicKey(signature->publicKey()));
-          if (validator_->validateChain(chain, *storage)) {
+          // Check chain last commit
+          auto is_chain_end_expected =
+              chain.last()
+                  .all([&commit_message](auto block) {
+                    return (commit_message->hash() == block->hash());
+                  })
+                  .as_blocking()
```

Comment:
Maybe `as_blocking()` then `last()`? So `first()` call isn't required

Reply:
can it be replaced with that?
`auto is_chain_end_expected = chain.as_blocking().last()->hash() == commit_message->hash();`